c++/src/capnp/rpc.c++

```diff
@@ -275,13 +293,13 @@ public:
   }
 
   T& nextOneWay(Id& id) {
-    // Choose an ID with the top bit set in round-robin fashion, but don't choose an ID that
+    // Choose an ID with the top bit(s) set in round-robin fashion, but don't choose an ID that
     // is still in use.
 
     uint startCounter = oneWayCounter;
     for (;;) {
-      id = oneWayCounter++ | ONE_WAY_BIT<Id>;
-      if (oneWayCounter == ONE_WAY_BIT<Id>) {
+      id = setQuestionIdType(oneWayCounter++, PIPELINE_ONLY);
+      if (oneWayCounter == SECOND_HIGHEST_BIT<Id>) {
```

Comment:
We're incrementing the `oneWayCounter`, and we want to wrap around as soon as it starts interfering with the enum bytes, which is why I changed the check to `SECOND_HIGHEST_BIT` instead of `ONE_WAY_BIT`.